src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -0,0 +1,297 @@
+        return String.join(
+                ",\n      ",
+                columns.stream()
+                    .map(c -> translateToJsonConvert(c.getName(), c.getType(), c.isArrayOf()))
+                    .collect(Collectors.toList()))
+            + "\n";
+      default:
+        throw new EnumConstantNotPresentException(FormatEnum.class, formatType.name());
+    }
+  }
+
+  private String buildWithStatement(FormatEnum formatType, DatasetTable datasetTable) {
+    switch (formatType) {
+      case CSV:
+        List<Column> columns = datasetTable.getColumns();
+        return String.join(
+            ",\n      ",
+            columns.stream()
+                .map(c -> c.getName() + " " + translateTypeToDdl(c.getType(), c.isArrayOf()))
+                .collect(Collectors.toList()));
+      case JSON:
+        return "doc nvarchar(max)";
+      default:
+        throw new EnumConstantNotPresentException(FormatEnum.class, formatType.name());
+    }
+  }
+
+  private String addArguments(FormatEnum formatType, int csvSkipLeadingRows) {
+    switch (formatType) {
+      case CSV:
+        return "PARSER_VERSION = '2.0'"
+            + ",\n                                "
+            + "FIRSTROW = "
+            + csvSkipLeadingRows;
+      case JSON:
+        return "fieldterminator ='0x0b'"
+            + ",\n                                "
+            + "fieldquote = '0x0b'";
+      default:
+        throw new EnumConstantNotPresentException(FormatEnum.class, formatType.name());
+    }
+  }
+
+  public void cleanSynapseEntries(
+      List<String> tableNames, List<String> dataSourceNames, List<String> credentialNames) {
+    tableNames.stream()
+        .forEach(
+            tableName -> {
+              try {
+                runAQuery("DROP EXTERNAL TABLE [" + tableName + "];");
+              } catch (Exception ex) {
+                logger.warn("Unable to clean up table {}, ex: {}", tableName, ex.getMessage());
+              }
+            });
+    dataSourceNames.stream()
+        .forEach(
+            dataSource -> {
+              try {
+                runAQuery("DROP EXTERNAL DATA SOURCE [" + dataSource + "];");
+              } catch (Exception ex) {
+                logger.warn(
+                    "Unable to clean up the external data source {}, ex: {}",
+                    dataSource,
+                    ex.getMessage());
+              }
+            });
+    credentialNames.stream()
+        .forEach(
+            credential -> {
+              try {
+                runAQuery("DROP DATABASE SCOPED CREDENTIAL [" + credential + "];");
+              } catch (Exception ex) {
+                logger.warn(
+                    "Unable to clean up scoped credential {}, ex: {}", credential, ex.getMessage());
+              }
+            });
+  }
+
+  public boolean runAQuery(String query) throws SQLException {
+    SQLServerDataSource ds = getDatasource();
+    try (Connection connection = ds.getConnection()) {
```

Comment:
I think you can do this in a single block:
```suggestion
    try (Connection connection = ds.getConnection();
         Statement statement = connection.createStatement())) {
```

Reply:
Ok - I attempted to do this. How does it look now?